src/controllers/user.js

```diff
@@ -12,7 +15,62 @@ const getProfile = async (req, res) => {
         .json(currentUser);
     }
   } catch (error) {
-    return res.sendStatus(500).json({ message: error.message });
+    return res.status(500).json({ message: error.message });
+  }
+};
+
+const getUserProducts = async (req, res) => {
+  try {
+    let { postType, SearchTitles } = req.query;
+    const filteringQueries = { $or: [] };
+    const finalQuery = { $and: [] };
+
+    if (postType) {
+      if (typeof postType !== 'object') postType = postType.split();
```

Comment:
what if postType is an Object? in that case the .forEach does not work correctly as expected